src/hook/useCalculate.ts

```diff
@@ -0,0 +1,61 @@
+import { calculrate, isNumberLengthValidate, isOperatorValidate } from "@/util";
+import { useState } from "react";
+
+const initialState = {
+  first: "",
+  second: "",
+  operator: "",
+  display: "",
+  computed: 0,
+};
+
+const useCalculate = () => {
+  const [calculrateState, setCalculrateState] = useState(initialState);
+  const { first, second, operator, display } = calculrateState;
+  const handleDigit = (digit: number) => {
+    if (!operator && isNumberLengthValidate(first)) {
+      setCalculrateState({
+        ...initialState,
+        first: first + digit,
+        display: display + digit,
+      });
+      return;
+    }
+
+    if (operator && isNumberLengthValidate(second)) {
+      setCalculrateState((prev) => ({
+        ...prev,
+        second: second + digit,
+        display: display + digit,
+      }));
+      return;
+    }
+  };
+  const handleOperation = (operation: string) => {
+    if (isOperatorValidate(first)) {
+      setCalculrateState((prev) => ({
+        ...prev,
+        operator: operation,
+        display: display + operation,
+      }));
+    }
+  };
+  const handleCalculate = () => {
+    const computed = calculrate({ first, second, operator });
+    setCalculrateState({ ...initialState, computed });
+  };
```

Comment:
```suggestion
  const computed = calculrate({ first, second, operator });
```
그냥 이렇게 표현하면 자연스럽게 계산되지 않을까요?
이런 것들은 굳이 state로 관리할 필요가 없어보여요.